src/bloomd/networking.c

```diff
@@ -254,11 +260,16 @@ static int setup_unix_listener(bloom_networking *netconf) {
         close(unix_listener_fd);
         return 1;
     }
-    if (listen(unix_listener_fd, BACKLOG_SIZE) != 0) {
+    if (listen(unix_listener_fd, UNIX_BACKLOG_SIZE) != 0) {
         syslog(LOG_ERR, "Failed to listen on UNIX socket! Err: %s", strerror(errno));
         close(unix_listener_fd);
         return 1;
     }
+    
+    // Setup the socket
+    if (set_non_blocking(unix_listener_fd)) {
```

Comment:
honestly that's not obvious without reading code, that there is difference between TCP and UNIX sockets  behaviour. probably better either make it nonblocking for both sockets (and add this to readme), or make it configurable? i.e. in config
```
unix_socket_nonblocking: true, 
tcp_socket_nonblocking: false
```

I just think it would be harder then to push to original repo

Reply:
I want actually to have a review of this change from armon. Because in redis there is no config for blocking mode, it just uses nonblocking mode for unix socket anyway so I think it some common thing. Config would probably be overengineering.

Reply:
then at least better have it consistent for both sockets?

Reply:
There are some conditional rules about setting non-blocking TCP socket in redis, probably needs some further research here
https://github.com/antirez/redis/blob/e07af6a2b782de809c1908a236332e48af5867b3/src/server.c#L1768